lib/index.js

```diff
@@ -290,9 +290,15 @@ exports.getAll = brCallbackify(async (
     actor,
     permission: PERMISSIONS.ACCOUNT_ACCESS
   });
-
+  // FIXME remove options.fields from all libraries that call on bedrock-account
+  // instead use options.projection
+  if(fields) {
+    logger.info('The parameter fields in method getAll is being deprecated' +
+      ' please use options.projection instead');
+  }
```

Comment:
```suggestion
```suggestion
  if(fields && options.projection) {
    throw new TypeError('"fields" or "options.projection" must be given, not both.');
  }
```

Reply:
@aljones15 This pattern was copied around to many modules.  It should be simplified to work with the `fields` param but error if trying to use both `fields` and `options.projection`.  The logger message would have been problematic as it would print out on every one of these calls.  (Using the `deprecate` module is a solution for that.)  To avoid breaking API changes, just accepting the fields param for now seems ok.  All the other modules with this pattern should be similarly updated.

Reply:
I do think we need the deprecation notice as the change from `fields` to `options.projections` is small, but will help. So I will look into using node's [util.deprecate](https://nodejs.org/api/util.html#util_util_deprecate_fn_msg_code) method. I was aware of it before, but because the arguments to the function have changed I skipped on using it at that time. I would also to add that printing this message out a lot of not a bad thing as the changes needed to update to the mongo 3.5 api are pretty small. ok so I will try to also throw if both fields and projections are passed in.

Reply:
@aljones15 -- we're not deprecating it. So we don't want a deprecation notice. We're just supporting a *new* feature of setting `projection` in the options. However, if you set `projection` in the options, you cannot also set `fields`, that is an error.

Reply:
I went ahead and went through each PR and corrected it to throw. I was using deprecation because obviously mongo no longer supports fields, but this is fine and it makes the upgrade easier. We should be ready for a release now, but deploying in production should be done as a test first.

Reply:
I think there's an issue with this.  The arg default of `fields = {}` means that to pass the test when you are using `object.projection` you would have to pass in `fields: null`.  That seems a bit odd.  Maybe use:
```js
  if(Object.keys(fields).length !== 0 && options.projection) {
```

Reply:
Maybe simpler to move the empty object default to later:
```js
  {actor, query = {}, fields = null, options = {}}) => {
```
```js
  if(fields && options.projection) {
```
```js
  options.projection = options.projection || fields || {};
```
Thoughts?

Reply:
I am going to go with making fields default to null and then adding the extra object. I have a dentist appointment this morning. Thanks for finding this and I will make the corrections once back from Dentist.